tools/src/main/java/org/apache/kafka/tools/ProducerPerformance.java

```diff
@@ -159,140 +180,205 @@ public static void main(String[] args) throws Exception {
+        public static final String TRANSACTIONAL_ID = "transactionalId";
+        public static final String TRANSACTION_DURATION_MS = "transactionDurationMs";
+
+        public String bootstrapServer() {
+            return this.ns.getString(BOOTSTRAP_SERVER);
+        }
+
+        public String topic() {
+            return this.ns.getString(TOPIC);
+        }
+
+        public long numRecords() {
+            return this.ns.getLong(NUM_RECORDS);
+        }
+
+        public int recordSize() {
+            return this.ns.getInt(RECORD_SIZE);
+        }
+
+        public String payloadFile() {
+            return this.ns.getString(PAYLOAD_FILE);
+        }
+
+        public String payloadDelimiter() {
+            return this.ns.getString(PAYLOAD_DELIMITER);
+        }
+
+        public int throughput() {
+            return this.ns.getInt(THROUGHPUT);
+        }
+
+        public List<String> producerProps() {
+            return this.ns.getList(PRODUCER_PROPS);
+        }
+
+        public String producerConfig() {
+            return this.ns.getString(PRODUCER_CONFIG);
+        }
+
+        public boolean printMetrics() {
+            return this.ns.getBoolean(PRINT_METRICS);
+        }
+
+        public String transactionalId() {
+            return this.ns.getString(TRANSACTIONAL_ID);
+        }
+
+        public long transactionDurationMs() {
+            return this.ns.getLong(TRANSACTION_DURATION_MS);
+        }
+
+        public ProducerPerformanceOptions(String[] args) throws ArgumentParserException {
+            super("producer-performance", "This tool is used to verify the producer performance.", args);
+        }
+
+        @Override
+        public void prepareArgs() {
+
+            MutuallyExclusiveGroup payloadOptions = this.parser
+                    .addMutuallyExclusiveGroup()
+                    .required(true)
+                    .description("either --record-size or --payload-file must be specified but not both.");
+
+            this.parser.addArgument("--bootstrap-server")
+                    .action(store())
+                    .type(String.class)
+                    .dest("bootstrapServer")
+                    .help("The server to connect to");
+
+            this.parser.addArgument("--topic")
+                    .action(store())
+                    .required(true)
+                    .type(String.class)
+                    .metavar("TOPIC")
+                    .help("produce messages to this topic");
+
+            this.parser.addArgument("--num-records")
+                    .action(store())
+                    .required(true)
+                    .type(Long.class)
+                    .metavar("NUM-RECORDS")
+                    .dest("numRecords")
+                    .help("number of messages to produce");
+
+            payloadOptions.addArgument("--record-size")
+                    .action(store())
+                    .required(false)
```

Comment:
Should we specify explicitly if this is the default?

Reply:
It's already there in the current (trunk) status of this tool. I prefer to leave it even for clarity (who don't know what the default is)

Reply:
I see, ok :)